main.go

```diff
@@ -76,25 +73,51 @@ func NewGame(g *Game) {
 	g.TileRenderer = renderer
 	g.LDTKProject = ldtkProject
 
+	level := g.LDTKProject.Levels[g.Level]
+
 	bg := ebiten.NewImage(
-		g.LDTKProject.Levels[g.Level].Width,
-		g.LDTKProject.Levels[g.Level].Height,
+		level.Width,
+		level.Height,
 	)
-	bg.Fill(g.LDTKProject.Levels[g.Level].BGColor)
+	bg.Fill(level.BGColor)
 
 	// Render map
-	g.TileRenderer.Render(g.LDTKProject.Levels[g.Level])
+	g.TileRenderer.Render(level)
 	for _, layer := range g.TileRenderer.RenderedLayers {
 		bg.DrawImage(layer.Image, &ebiten.DrawImageOptions{})
 	}
 	g.Background = bg
 
-	//Load sprites
+	// Create space for collision detection
+	g.Space = resolv.NewSpace(level.Width, level.Height, 16, 16)
+
+	// Add wall tiles to space for collision detection
+	for _, layer := range level.Layers {
+		switch layer.Type {
+		case ldtkgo.LayerTypeIntGrid:
+
+			if ints := layer.IntGrid; len(ints) > 0 {
```

Comment:
This code will probably run fine without this if statement too, because the for loop will execute zero times, i.e. the for-loop's statement block won't even be run if ints is zero, so the result is the same.

Reply:
You are right, I change it